src/Lykke.Service.BlockchainApi.Sdk/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Autofac;
+using Lykke.AzureStorage.Tables.Entity.Metamodel;
+using Lykke.AzureStorage.Tables.Entity.Metamodel.Providers;
+using Lykke.Common.Log;
+using Lykke.Service.BlockchainApi.Sdk.Controllers;
+using Lykke.Service.BlockchainApi.Sdk.Domain;
+using Lykke.Service.BlockchainApi.Sdk.PeriodicalHandlers;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Lykke.Service.BlockchainApi.Sdk
+{
+    public static class ServiceCollectionExtensions
+    {
+        static IServiceCollection AddBlockchainControllers(this IServiceCollection services,
+            Type[] controllerTypes)
+        {
+            services = services ??
+                throw new ArgumentNullException(nameof(services));
+
+            controllerTypes = controllerTypes ??
+                throw new ArgumentNullException(nameof(controllerTypes));
+
+            var apmInstance = services.FirstOrDefault(s => s.ServiceType == typeof(ApplicationPartManager))?.ImplementationInstance;
+            var apm = apmInstance as ApplicationPartManager ??
+                throw new ArgumentException("ApplicationPartManager not found");
+
+            // we will add only specific controllers,
+            // so remove SDK application part first
+            var sdk = apm.ApplicationParts.FirstOrDefault(part => part.Name == Assembly.GetExecutingAssembly().GetName().Name);
```

Comment:
It's better to cache Assembly.GetExecutingAssembly().GetName().Name in local var, to avoid Reflection usage on each item processing

Reply:
Fixed.